### ISSUE TITLE
[QUESTION] Is possible to publish and consume messages in Aws MSK with SASL enabled?

### ISSUE THREAD

**Issue by daavelar** [mateusjunges/laravel-kafka#96]
Hi, i'm trying to publish a message to an AWS MSK broker using SASL authentication with SCRAM mechanism, but i'm getting the message "Sent messages may not be completed yet." as response. Below you can see my code:

```php
      $message = new Message(
            body: $request->get('message'),
        );

        $sasl = new Sasl(
            username: 'myuser',
            password: 'mypass',
            mechanisms: 'SCRAM-SHA-256'
        );

        try {
            Kafka::publishOn('test')
                ->withSasl($sasl)
                ->withMessage($message)
                ->send();
        } catch (CouldNotPublishMessage $exception) {
            return back()->with('error', $exception->getMessage());
        }
```

Am I missing something or there is some new feature needed to publish in AWS MSK?

**Comment by rtuin** [comment]
@daavelar Publishing (and) to MSK with SASL/SCRAM works fine. 
Not sure if directly related, but as far as I know MSK uses `SCRAM-SHA-512`. The error message would be confusing if this is the problem you're facing, though.

**Comment by daavelar** [comment]
I got it publishing the message, but for consuming i'm still having an issue, i'm using KafkaConsumerCommand and i have included the sasl and securityProtocol info in the config file:

**Disconnected while requesting ApiVersion: might be caused by incorrect security.protocol configuration (connecting to a SSL listener?) or broker version is < 0.10 (see api.version.request) (after 159ms in state APIVERSION_QUERY)**

**Comment by daavelar** [comment]
I think I will need to build the kafka extension with the SSL support enabled:
https://github.com/confluentinc/confluent-kafka-go/issues/411

**Comment by daavelar** [comment]
I found out here what the problem was, even with the flag --securityProtocol=SASL_SSL, the consumer was using the default SASL_PLAINTEXT value because the parameter was missing in the getSasl() method of the Options class, i change the method to this and worked:
```php
    public function getSasl(): ?Sasl
    {
        if (is_null($this->saslMechanisms) || is_null($this->saslPassword) || is_null($this->saslUsername)) {
            return null;
        }

        return new Sasl(
            username: $this->saslUsername,
            password: $this->saslPassword,
            mechanisms: $this->saslMechanisms,
            securityProtocol: $this->securityProtocol
        );
    }
```

**Comment by mateusjunges** [comment]
@daavelar can you make a PR?

**Comment by mateusjunges** [comment]
@daavelar fixed on v1.7.5. Please let me know if it solves your issue

**Comment by daavelar** [comment]
Solved! Thanks a lot!
